Launcher/Classes/ViewControllers/TasksViewController.swift

```diff
@@ -352,50 +306,54 @@ class TasksViewController: NSViewController {
         }
     }
     
-    func disableBuildItems() {
-        if simulatorRadioButton.state == .on {
-            buildPicker.item(at: 0)?.isEnabled = true
-            buildPicker.item(at: 1)?.isEnabled = false
-            buildPicker.item(at: 2)?.isEnabled = true
-        } else {
-            buildPicker.item(at: 0)?.isEnabled = false
-            buildPicker.item(at: 1)?.isEnabled = true
-            buildPicker.item(at: 2)?.isEnabled = false
-        }
-        
-        if buildPicker.selectedItem?.isEnabled == false {
-            cautionBuildImage.isHidden = false
-            buildItemIsDisabled = true
-        } else {
-            cautionBuildImage.isHidden = true
-            buildItemIsDisabled = false
-        }
-    }
-    
-    func getSimulators() {
-        spinner.startAnimation(self)
-        progressBar.startAnimation(self)
-        getDeviceButton.isEnabled = false
-        isRunning = true
-        
+    func getSimulatorsCommand() {
         if let launchPath = Constants.FilePaths.Bash.simulators {
             let outputStream = CommandsCore.CommandTextOutputStream()
             outputStream.textHandler = { text in
+                let filderedText = text.components(separatedBy: "\n").filter { !$0.isEmpty }
                 DispatchQueue.main.async {
-                    let filderedText = text.components(separatedBy: "\n").filter { !$0.isEmpty }
                     self.phoneComboBox.addItems(withTitles: filderedText)
                 }
             }
-            commands.executeCommand(at: launchPath, arguments: [], outputStream: outputStream)
+            self.commands.executeCommand(at: launchPath, arguments: [], outputStream: outputStream)
+        }
+    }
+    
+    func emptyDeviceHandler() {
```

Comment:
I think we can improve this function's readability by handling the "happy" path first:

```swift
if let _ = phoneComboBox.selectedItem {
  self.cautionImage.isHidden = true
  self.deviceListIsEmpty = false
} else {
  self.deviceListIsEmpty = true
  self.phoneComboBox.highlight(true)
  self.cautionImage.isHidden = false
  let noConnectedDevices = "\(Constants.Strings.noDevicesConnected) \(Constants.Strings.installSimulatorOrPluginDevice)"
  self.phoneComboBox.addItem(withTitle: noConnectedDevices)
  let previousOutput2 = self.textView.string
  self.textView.string = "\(previousOutput2)\n\(noConnectedDevices)"
}
```

Or maybe even better: add a variable that checks if an item is selected:

```swift
var isDeviceListEmpty: Bool {
  return phoneComboBox.numberOfItems == 0
}
```

... which is also more true - the list is empty when the list is empty, not when there's no item selected. :)